utils/exception.py

```diff
@@ -0,0 +1,11 @@
+from rest_framework.exceptions import APIException
+from rest_framework import status
+
+class CustomException(APIException):
+    status_code = status.HTTP_400_BAD_REQUEST
+    default_code = 'error'
+
+    def __init__(self, detail, status_code=None):
+        self.detail = detail
+        if status_code is not None:
+            self.status_code = status_code
```

Comment:
I can't see any benefit of creating a custom exception. If our sole purpose is raising `403` errors then we can simply use `PermissionError` instead of `ValidationError`

/cc @krashish8 what do you think

Reply:
Yeah, there isn't any benefit of creating a custom exception, unless we want to tweak the response body of the error response.

E.g. Custom exception can be used if we always want the HTTP status code to be included in the body of the response, i.e. ```{"status_code": 401, "detail": "Authentication credentials were not provided."}``` instead of
`{"detail": "Authentication credentials were not provided."}`

@aishwary023, you have implemented the same [`ValidationError` class defined in django-rest-framework](https://github.com/encode/django-rest-framework/blob/6a45a6a36fedf5b2a9bfbb23cf12b8a1cc8acb3e/rest_framework/exceptions.py#L138-L154), so instead of that, you could have directly used `ValidationError`.

Reply:
@nishantwrp Will it be better if we also display "status_code" in the response body? Or is it just unnecessary?

Reply:
The frontend can access the status code anyways. So, l don't think we need an extra `status_code` field.